services/motion/builtin/builtin.go

```diff
@@ -430,3 +394,87 @@ func (ms *builtIn) GetPose(
 		supplementalTransforms,
 	)
 }
+
+// PlanMoveOnMap returns the plan for MoveOnMap to execute
+func (ms *builtIn) planMoveOnMap(
+	ctx context.Context,
+	componentName resource.Name,
+	destination spatialmath.Pose,
+	slamName resource.Name,
+	extra map[string]interface{},
+) ([][]referenceframe.Input, base.KinematicBase, error) {
+	// get the SLAM Service from the slamName
+	localizer, ok := ms.localizers[slamName]
+	if !ok {
+		return nil, nil, resource.DependencyNotFoundError(slamName)
+	}
+
+	// assert localizer as a slam service and get map limits
+	slamSvc, ok := localizer.(slam.Service)
+	if !ok {
+		return nil, nil, fmt.Errorf("cannot assert localizer of type %T as slam service", localizer)
+	}
+
+	// gets the extents of the SLAM map
+	limits, err := slam.GetLimits(ctx, slamSvc)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// create a KinematicBase from the componentName
+	component, ok := ms.components[componentName]
+	if !ok {
+		return nil, nil, resource.DependencyNotFoundError(componentName)
+	}
+	kw, ok := component.(base.KinematicWrappable)
+	if !ok {
+		return nil, nil, fmt.Errorf("cannot move component of type %T because it is not a KinematicWrappable Base", component)
+	}
+	kb, err := kw.WrapWithKinematics(ctx, localizer, limits)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pointCloudData, err := slam.GetPointCloudMapFull(ctx, slamSvc)
+	if err != nil {
+		return nil, nil, err
+	}
+	octree, err := pointcloud.ReadPCDToBasicOctree(bytes.NewReader(pointCloudData))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if extra == nil {
+		extra = make(map[string]interface{})
+	}
+	extra["planning_alg"] = "rrtstar"
```

Comment:
I want to make sure that the planning algorithm is set to rrtstar and not cbirrt. I guess this would mean that it overrides any other planning algs that may have been set. I can change this if you think it's best to not potentially override the "planning_alg" field ever

Reply:
> I want to make sure that the planning algorithm is set to rrtstar and not cbirrt

Why?

Reply:
Yeah I think we want to be defaulting to this algo for now.  A lot is going to be changing quickly here so I think this is fine to do for now

Reply:
It's correct that cbirrt will create paths composed of larger quantities of waypoints, yes.

Why are large numbers of waypoints a problem? Is this just an artifact of the current MoveStraight/Spin implementation that will cause stops frequently?

Reply:
Yeah it made it really difficult to be followed by our wrapper.  The arms can just move between the intermediate waypoints but there are spin moves in between each waypoint so it resulted in really weird looking behavior.  Quality improved a lot when switching to RRT*